sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -503,14 +553,27 @@ static int sof_audio_probe(struct platform_device *pdev)
 		dmic_num = 0;
 		hdmi_num = 0;
 		/* default quirk for legacy cpu */
-		sof_rt5682_quirk = SOF_RT5682_SSP_CODEC(2);
+		sof_rt5682_quirk = SOF_RT5682_MCLK_EN |
+						SOF_RT5682_MCLK_BYTCHT_EN |
+						SOF_RT5682_SSP_CODEC(2);
```

Comment:
sorry, this makes no sense. You set the default quirk *after* testing it.
That leads me to think you haven't testing this code on actual MinnowBoard Turbot?

Reply:
I tested on MinnowBoard Turbot because I have this platform only. This is my platform name and it works on it:

> Minnowboard Turbot D0 PLATFORM

For other BYT/CHT, I have not tested because I have no device. What do you mean by broken? It does not work on other BYT/CHT device?

Reply:
please recheck the flow, you haven't setup the default quirk but you test it at the beginning of the function. This cannot possibly work.

Reply:
Do you mean the line 210 for the "test"? `if (sof_rt5682_quirk & SOF_RT5682_MCLK_BYTCHT_EN) {`

Reply:
I think we have a misunderstanding caused by github's auto omission here. Some lines are omited by default. Line 210 and line 556 actually are not in the same function. One in `sof_rt5682_hw_params` and another in `sof_audio_probe`.

Reply:
ah right, this looks ok, but you have two additional problems with this machine driver:
a) there is no need to use GFP_ATOMIC. @bardliao is there any reason why this was used?
b) you need to use the helpers I added upstream to remove the dependencies on asm files and x86_match_cpu. See https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-intel-quirks.h

Reply:
> a) there is no need to use GFP_ATOMIC. @bardliao is there any reason why this was used?

No, I can't remember why I use GFP_ATOMIC at all. And I think the device can sleep. @xun2z Could you send another commit to change it to GFP_KERNEL?

Reply:
please discard my comment b), I thought the code was merged upstream but it was still in RFC state. we should fix the GFP_ATOMIC with an additional commit and that'll do for now. Thanks @xun2z!

Reply:
There was a fix submitted upstream minutes ago for this so we are good to go with this PR.

Reply:
When I'm trying to add the helper, I also noticed this, yes I should add a commit to change GFP_ATOMIC